### DISCUSSION TITLE
Heap out of memory

### DISCUSSION THREAD

**Discussion by P4sca1** [nuxt/bridge#394] [General]
Running `nuxi build` even without pre-rendering takes up a huge amount of memory (up to 4GB) to build the page. The webpack build works fine and takes around 1GB. But as soon as the nitro server build starts, the memory usage is skyrocketing.
Because of that I can’t build my nuxt application in environments with limited memory constraints like GitHub Actions or in a Docker container. 

I can’t provide a reproduction, so I started a discussion. What is your experience with nuxt bridge memory usage?

**Comment by danielroe** [comment]
I would recommend turning source maps off for your nitro build. You should be able to configure in your Nuxt config under the nitro key.

**Reply by P4sca1** [reply]
I am using the nuxt top level option `sourcemap: false`. Do I need to disable source map generation separately for nitro?

**Reply by danielroe** [reply]
Yes. I don't believe the top level option is implemented yet for Bridge.

**Reply by P4sca1** [reply]
Setting `sourceMap: false` in the nitro configuration did not help.

**Comment by P4sca1** [comment]
I was able to reduce the memory usage by more than 1GB.
Previously, I imported icons from the FontAwesome 6 icon library via
```ts
import { IconA, IconB } from '@fortawesome/free-solid-svg-icons'
```
This resulted in the entire icon library (around 3.200 icons per preset, I was using 2 presets) to be inlined into the `.output/server/chunks/app/server.mjs` file, which has grown to a size of more than 9MB.
I expected only used icons to be included in the file, which did not work for some reason.
After updating my code to import icons differently, only icons that are used get imported in the `server.mjs` file and the icons are not written into the `server.mjs` file directly (but are imported).
```ts
import { IconA } from '@fortawesome/free-solid-svg-icons/IconA'
import { IconB } from '@fortawesome/free-solid-svg-icons/IconB'
```

I think the memory usage is still quite high and I am wondering, why tree shaking wont remove unused icons.

**Comment by AndreyYolkin** [comment]
This is what I am talking about here: https://github.com/nuxt/bridge/discussions/379 , will try to disable sourceMaps, too

**Comment by JuliWolf** [comment]
Have the same issue. 

Starts after updating nuxt to version Nuxt CLI v3.0.0-rc.6  

Client part build successfully,  webpack generates server chunk and then throws error `Javascript Heap out of memory`

set `sourceMap` to false did not help

**Comment by danielroe** [comment]
Would you try https://github.com/nuxt/bridge/issues/421#issuecomment-1189152999 and see if this helps?

**Reply by AndreyYolkin** [reply]
I tried it and build time drastically reduced!
Heap usage reduced to 6Gb, too. It's a big, but this is related with [rollup keeping information strategy](https://rollupjs.org/guide/en/#error-javascript-heap-out-of-memory) as far as I know
